settings.gradle.kts

```diff
@@ -1 +1,45 @@
+import org.gradle.api.initialization.resolve.RepositoriesMode.FAIL_ON_PROJECT_REPOS
+
+plugins {
+    id("org.gradle.toolchains.foojay-resolver-convention") version ("0.9.0")
+}
+
+dependencyResolutionManagement {
```

Comment:
Why do we need the changes in this file? It's related to the question "why adding the JS target?".

Reply:
The foojay resolver convention plugin always makes sense if you use JVM toolchains which you also did before, otherwise you cannot (or will not be able to in the future) auto-provision the needed toolchains.

The repositories are for the JS part of KMP as I greatly dislike plugins to add repositories to projects as that opens much wider the doors to supply chain attacks, so disabled the repository adding and added the necessary ones here.

The version catalog is for the Kotlin Wrappers for the JS libs.